### ISSUE TITLE
Hugo 0.38, Panic invalid URL escape "% e"

### ISSUE THREAD

**Issue by brunoamaral** [gohugoio/hugo#4586]
This is from a site that was working in version 0.37.1, now it returns the error bellow:

`Building sites … panic: parse /edpbrasil/2015-10-28 - EBITDA da EDP avança 84,7% e chega a R$ 586,9 milhões no 3º tri/images/image.jpg: invalid URL escape "% e"`

**Comment by anthonyfok** [comment]
Hi @brunoamaral!  Thank you for your report!

Hugo v0.38.1 (bugfix) was just released yesterday.  Could you please try the newer version and see if you could reproduced the problem you are seeing?  Many thanks!

**Comment by brunoamaral** [comment]
Just tried to reproduce using 0.38.1 and it's working as expected. Between posting and trying 0.38.1, I got the build to work by changing the folder name and removing the % sign.

**Comment by bep** [comment]
>Just tried to reproduce using 0.38.1 and it's working as expected. 

Do you say that this is fixed in 0.38.1?

**Comment by sa3dany** [comment]
I was migrating content into a hugo site on v0.54.0/extended and one of the images had a "%u" in the name. When building I get:

```
Building sites … panic: parse /ar/men%u2019s-collection.jpg: invalid URL escape "%u2"
```

```
goroutine 173 [running]:
github.com/gohugoio/hugo/helpers.(*PathSpec).URLEscape(0xc0001259b0, 0xc000ba8690, 0x4c, 0x4c, 0x4c)
        /go/src/github.com/gohugoio/hugo/helpers/url.go:121 +0x8e
github.com/gohugoio/hugo/helpers.(*PathSpec).URLizeFilename(0xc0001259b0, 0xc000ba8690, 0x4c, 0xc001010000, 0xc000ba8690)
        /go/src/github.com/gohugoio/hugo/helpers/url.go:112 +0x5e
github.com/gohugoio/hugo/resources.(*genericResource).relPermalinkForRel(0xc0001c7ce0, 0xc000141029, 0x48, 0x0, 0xc000141029, 0x48)
        /go/src/github.com/gohugoio/hugo/resources/resource.go:572 +0x89
github.com/gohugoio/hugo/resources.(*genericResource).relPermalinkFor(0xc0001c7ce0, 0xc000141029, 0x48, 0x48, 0xc000141029)
        /go/src/github.com/gohugoio/hugo/resources/resource.go:522 +0x4b
github.com/gohugoio/hugo/resources.(*genericResource).RelPermalink(0xc0001c7ce0, 0x0, 0x0)
        /go/src/github.com/gohugoio/hugo/resources/resource.go:514 +0x84
github.com/gohugoio/hugo/hugolib.(*contentHandlers).parsePage.func1.1(0x1, 0x0, 0xc001451bd0)
        /go/src/github.com/gohugoio/hugo/hugolib/pagebundler_handlers.go:254 +0x1b7
sort.insertionSort_func(0xc001451cc8, 0xc0015610b0, 0x0, 0x4)
        /usr/local/go/src/sort/zfuncversion.go:12 +0x9a
sort.stable_func(0xc001451cc8, 0xc0015610b0, 0x4)
        /usr/local/go/src/sort/zfuncversion.go:171 +0x8d
sort.SliceStable(0x112b9a0, 0xc00138b8a0, 0xc001451cc8)
        /usr/local/go/src/sort/slice.go:31 +0x10d
github.com/gohugoio/hugo/hugolib.(*contentHandlers).parsePage.func1(0xc000d9df80, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/gohugoio/hugo/hugolib/pagebundler_handlers.go:238 +0x701
github.com/gohugoio/hugo/hugolib.(*contentHandlers).processFirstMatch.func1(0xc000d9df80, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/gohugoio/hugo/hugolib/pagebundler_handlers.go:85 +0x95
github.com/gohugoio/hugo/hugolib.(*siteContentProcessor).readAndConvertContentBundle(0xc0005ef770, 0xc0007b2a40, 0x2, 0x0)
        /go/src/github.com/gohugoio/hugo/hugolib/pagebundler.go:208 +0x71
github.com/gohugoio/hugo/hugolib.(*siteContentProcessor).process.func4(0x8, 0x128e0a8)
        /go/src/github.com/gohugoio/hugo/hugolib/pagebundler.go:172 +0x121
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0005cf200, 0xc000470a60)
        /go/pkg/mod/golang.org/x/sync@v0.0.0-20180314180146-1d60e4601c6f/errgroup/errgroup.go:58 +0x5e
created by golang.org/x/sync/errgroup.(*Group).Go
        /go/pkg/mod/golang.org/x/sync@v0.0.0-20180314180146-1d60e4601c6f/errgroup/errgroup.go:55 +0x6d
```

And the relevant function call:
https://github.com/gohugoio/hugo/blob/5df2b79dd2734e9a00ed1692328f58c385676468/helpers/url.go#L118

I am not familiar with Go but I googled that error and learned about ```url.QueryEscape(s)```  which escapes the string (in this case replacing the "%" with "%25") so that ```url.Parse()``` would not panic. But, ```url.QueryEscape(s)``` also escaped the path separator in my tests (as returned by the u.String() method).
